module/modmanager/manager_test.go

```diff
@@ -1322,3 +1323,34 @@ func TestBadModuleFailsFast(t *testing.T) {
 
 	test.That(t, err.Error(), test.ShouldContainSubstring, "module test-module exited too quickly after attempted startup")
 }
+
+func TestModuleDiscoverRegistered(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	// Precompile module to avoid timeout issues when building takes too long.
+	modPath := rtestutils.BuildTempModule(t, "examples/customresources/demos/complexmodule")
```

Comment:
```suggestion
	modPath := rtestutils.BuildTempModule(t, "module/testmodule")
```

What you have is totally valid, but I suggest we use the testmodule here. We can extend the test module ([source](https://github.com/viamrobotics/rdk/tree/main/module/testmodule)) to actually have a `Discover` method that returns something we can assert against in this unit test. testmodule isn't used in any user-facing examples so we are free to extend it as we please.